rust/blockstore/src/arrow/types.rs

```diff
@@ -18,13 +18,20 @@ pub trait ArrowWriteableKey: Key + Default {
 }
 
 pub trait ArrowWriteableValue: Value {
```

Comment:
I think we should comment all the GATs and methods now since this type has gotten rather large (good thing)